app/jobs/yearly_review.rb

```diff
@@ -103,23 +103,28 @@ def category_post_topics(cat_id, start_date, end_date)
     end
 
     def review_categories_from_settings
+      read_restricted = SiteSetting.yearly_review_include_private_categories
+      opts = {}
+      opts[:read_restricted] = false if read_restricted == false
       if SiteSetting.yearly_review_categories.blank?
-        Category.where(read_restricted: false).order("topics_year DESC")[0, 5].pluck(:id)
+        Category.where(opts).order("topics_year DESC")[0, 5].pluck(:id)
```

Comment:
I think the plugin has always limited the categories that are used in the topics that it features to public topics. There was an issue that private categories were used for the stats section that was fixed last year. My concern is that the categories returned by this setting are used for determining which topics are displayed by the `create_category_posts` method. Maybe read restricted categories should have to be explicitly included in the `yearly review categories` setting for them to be used?

Reply:
WIth that approach, the `read_restricted` setting would only be applied if the `yearly_review_categories` setting has been set.